xbmc/utils/SysfsUtils.cpp

```diff
@@ -16,105 +16,140 @@
 #include <fcntl.h>
 #include <string.h>
 
-#ifdef TARGET_WINDOWS_STORE
-#include <io.h>
-#endif
+static const unsigned int PAGESIZE = getpagesize();
 
-int SysfsUtils::SetString(const std::string& path, const std::string& valstr)
+bool SysfsUtils::SetString(const std::string& path, const std::string& value)
 {
-  int fd = open(path.c_str(), O_RDWR, 0644);
-  int ret = 0;
-  if (fd >= 0)
+  auto oldValue = GetString(path);
+  if (oldValue == value)
   {
-    if (write(fd, valstr.c_str(), valstr.size()) < 0)
-      ret = -1;
-    close(fd);
+    return true;
+  }
+
+  if (value.size() > PAGESIZE)
+  {
+    CLog::Log(LOGERROR, "{}: string length larger than page size {}", __FUNCTION__, PAGESIZE);
+    return false;
+  }
+
+  int fd = open(path.c_str(), O_RDWR);
+  if (fd < 0)
+  {
+    CLog::Log(LOGERROR, "{}: error opening {}", __FUNCTION__, path);
```

Comment:
I think with `SetString: error opening xxxxx` the `SetString: ` prefix is not going to be especially helpful. Rather write something descriptive (sysfs utils or the like).